### DISCUSSION TITLE
webgis.tools.serialization - Dateinamen

### DISCUSSION THREAD

**Discussion by Andreas-Lampert** [e-netze/webgis-community#204] [Q&A]
Hallo Jürgen!

Für die Umstellung des NÖ Atlas von WebGIS5 auf WebGIS6 wollte ich die Konfig-Dateien am Fileshare in einen neuen Ordner WebGIS6 kopieren. 

Im NÖ Atlas bieten wir den Usern die Möglichkeit Karten zu speichern. Diese werden ja im Ordner webgis.tools.serialization gespeichert (ich hoffe ich habe das richtig verstanden). 
Beim Abspeichern wird im _anonym$...-Ordner eine hex-....blb und ...-graphics.dat-Datei angelegt. Dieser Dateiname wird bei ca. 1 % aller Ordner so lange, dass ich den Ordner nicht mehr kopieren oder verschieben kann. Der Pfad ist zu lange (siehe Screenshot).

![image](https://github.com/user-attachments/assets/7de06291-6789-4650-a749-723738c09c80)

Was genau wird in diesem Dateinamen hinterlegt?
Könntest du eine Logik einbauen, um den Dateinamen zu kürzen bzw. eine Begrenzung einbauen, um so lange Bezeichnungen zu unterbinden. 

Danke!

Lg, Andi

**Comment by gis-eni** [comment]
Im Filename steckt der Titel des "Projekts". Eine Idee wäre die Länge des Titels einzuschränken

**Reply by ChristophNeumayrNOE** [reply]
Hallo Jürgen. 
Ich bin nun leider über das gleiche Problem mit den zu langen Dateinamen gestoßen.
Wäre es bitte möglich, wie vorgeschlagen, die Länge des Titels einzuschränken.
Das würde uns sehr helfen.
LG
Christoph

**Reply by gis-eni** [reply]
Die Frage ist halt, worauf einschränken?

**Reply by ChristophNeumayrNOE** [reply]
Mein Vorschlag wären 30 Zeichen? Dann wäre die Datei rund 77 Zeichen lang.

**Reply by gis-eni** [reply]
Laut Doku gibt es dazu eh schon einen Eintrag für die ``api.config``:

https://docs.webgiscloud.com/de/webgis/config/api/index.html#werkzeug-karte-speichern

``` xml
<section name="tool-savemap">
  <add key="name-maxlength" value="40" />
</section>
```

Habt ihr das schon ausprobiert? Default ist 40 Zeichen.

**Reply by gis-eni** [reply]
Das Problem kann übrigens auch auftreten, wenn Zeichnungen im MapMarkup Tool gespeichert werden.

Dazu gibt ab 8.26.1001 ebenfalls einen neuen Key:

``` xml
<section name="tool-mapmarkup">
         <!-- other settings -->
	 <add key="save-name-maxlength" value="20" />  <!-- default: 40 -->
</section>
```

<img width="1804" height="754" alt="image" src="https://github.com/user-attachments/assets/867fa416-f89c-4f2a-aa9b-2ecbc8b4db8e" />

**Comment by Andreas-Lampert** [comment]
Workaround: Mounten des Netzwerkpfades als Netzlaufwerk bis z.B. ...\config\portal\storage. Damit wird der Dateipfad kurz genug und es können die Ordner+Dateien kopiert werden.
